stdlib/http/src/main/java/org/ballerinalang/net/http/WebSocketObservability.java

```diff
@@ -0,0 +1,387 @@
+                observerContext.addTag(TAG_CONNECTION_ID, WEBSOCKET_UNKNOWN);
+            }
+            setObserveService(observerContext, connectionInfo);
+            observerContext.addTag(TAG_ERROR_TYPE, errorType);
+            //If the error is related to sending/receiving a message, set the type of message
+            if (messageType != null) {
+                observerContext.addTag(TAG_MESSAGE_TYPE, messageType);
+            }
+            Map<String, String> tags = observerContext.getTags();
+            Set<Tag> allTags = new HashSet<>(tags.size());
+            Tags.tags(allTags, observerContext.getTags());
+
+            MetricRegistry metricRegistry = DefaultMetricRegistry.getInstance();
+            //Increment errors metric
+            metricRegistry.counter(new MetricId(SERVER_CONNECTOR_WEBSOCKET + "_" + METRIC_ERRORS,
+                                                METRIC_ERRORS_DESC, allTags)).increment();
+
+            //Log error
+            if (messageType == null) {
+                logger.error("type:{}, message: {}, connectionId: {}, service:{}",
+                             errorType, errorMessage, getConnectionId(connectionInfo), getService(connectionInfo));
+            } else {
+                logger.error("type:{}/{}, message: {}, connectionId: {}, service:{}",
+                             errorType, messageType, errorMessage, getConnectionId(connectionInfo),
+                             getService(connectionInfo));
+            }
+        }
+    }
+
+    /**
+     * This method determines whether the current context is server or client and sets the tags accordingly.
+     * If server, the service bas path.
+     * If client, the remote URL
+     *
+     * @param observerContext current observer context
+     * @param connectionInfo information regarding connection.
+     */
+    private static void setObserveService(ObserverContext observerContext, WebSocketOpenConnectionInfo connectionInfo) {
+        try {
+            String service = connectionInfo.getService().getBasePath();
+            if (service != null) {
+                //If base path is set (i.e. server)
+                observerContext.addTag(TAG_CLIENT_OR_SERVER, WEBSOCKET_CLIENT_OR_SERVER_SERVER);
+                observerContext.addTag(TAG_SERVICE, service);
+            } else {
+                //if base path is not set (i.e. client)
+                observerContext.addTag(TAG_CLIENT_OR_SERVER, WEBSOCKET_CLIENT_OR_SERVER_CLIENT);
+                observerContext.addTag(TAG_SERVICE, connectionInfo.getWebSocketEndpoint().getStringValue("url"));
+            }
+        } catch (NullPointerException e) {
+            observerContext.addTag(TAG_CLIENT_OR_SERVER, WEBSOCKET_UNKNOWN);
+            observerContext.addTag(TAG_SERVICE, WEBSOCKET_UNKNOWN);
+        }
+
+    }
+
+    private static String getService(WebSocketOpenConnectionInfo connectionInfo) {
+        try {
+            String service = connectionInfo.getService().getBasePath();
```

Comment:
Note that service is not the path
```suggestion
            String path = connectionInfo.getService().getBasePath();
```